### PR TITLE
Support multiple date selection

### LINKED ISSUE #43: Multiple date support
image

Hey @RichardLindhout! Awesome library. Is there a way to select multiple dates like the screenshot above?

### PR DESCRIPTION
**Pull request by nandorojo**
This PR plans to add multiple date support, to close #43.

- [x] Add `dates` prop to `Calendar`, allowing it to receive multiple dates rather than just 1
- [x] Add `mode="multiple"` to `DatePicker`
- [x] Add label to `DatePickerModalHeaderContent` when selecting multiple dates
- [x] ~~Add option to require at least 1 date (i.e. don't remove the last date if it's changed.) Maybe this should just be done by users by not changing the date if `length < 1`~~ Instead, we notify users with the day just clicked and whether it was added or removed 

@RichardLindhout Can you confirm that we can close this if I can get it done and make it work well?

Thanks!

### CODE REVIEW

**Comment by nandorojo** [comment]
All set, looks good to go. All added to the example too.

Here's a video: https://www.loom.com/share/ea24599c192041d5abbdedb580e136c0

**Comment by RichardLindhout** [comment]
Thanks for al your work!! <3

I'll get back in a few hours to accept it and make a new release!

**Comment by nandorojo** [comment]
No problem! I have one last change to commit and it should be good to go. It's not blocking, but it's an added feature. When you call `onChange`, in addition to `dates`, it should also give you the date you just clicked and whether it was added or removed.

File: src/Date/Calendar.tsx [hunk 1]
```diff
@@ -143,9 +158,21 @@ function Calendar(
...
+        datesRef.current = datesRef.current || []
+        const exists = datesRef.current?.some((ed) => areDatesOnSameDay(ed, d))
+
+        const newDates = exists
+          ? datesRef.current.filter((ed) => !areDatesOnSameDay(ed, d))
+          : [...datesRef.current, d]
+
+        newDates.sort((a, b) => a.getTime() - b.getTime())
+        ;(onChangeRef.current as MultiChange)({
+          dates: newDates,
```
Comment (nandorojo) [inline src/Date/Calendar.tsx:171]: We should add `pressed: Date, removed: boolean` here.

**Comment by nandorojo** [comment]
Good to go

**Comment by RichardLindhout** [comment]
Really happy with it! You are amazing

**Comment by RichardLindhout** [comment]
Is it ok to add your video to the release for people to see it in action?

**Comment by RichardLindhout** [comment]
I did one slight API change renamed "multi" to "multiple"

**Comment by nandorojo** [comment]
Yeah feel free to share the video!

`multiple` sounds good, too.

**Comment by RichardLindhout** [comment]
I did already :)

**Comment by RichardLindhout** [comment]
https://github.com/web-ridge/react-native-paper-dates/releases/tag/v0.3.0

**Comment by iM-GeeKy** [comment]
@nandorojo Hey I messaged you on discord, but ultimately just wanted to pick your brain on an issue related to this feature (https://github.com/web-ridge/react-native-paper-dates/issues/383). I'm working on trying to fix this!

**Pull request merged**